cdap-ui/app/directives/timeline/timeline-ctrl.js

```diff
@@ -24,7 +27,73 @@ function TimelineController ($scope, LogViewerStore, LOGVIEWERSTORE_ACTIONS) {
       }
     });
   };
+
+  var pollPromise = null;
+
+  var apiSettings = {
+    metric : {
+      context: `namespace.${this.namespaceId}.app.${this.appId}.flow.${this.programId}.run.${this.runId}`,
+      names: ['system.app.log.error', 'system.app.log.warn', 'system.app.log.info', 'system.app.log.debug'],
+      startTime : '',//0,
+      endTime : '',
+      resolution: '1m'
+    }
+  };
+
+  function pollForMetadata() {
+
+    console.log('Before Poll: start: ', apiSettings.metric.startTime);
+    console.log('end: ', apiSettings.metric.endTime);
+    pollPromise = dataSrc.poll({
+      _cdapPath: '/metrics/query',
+      method: 'POST',
+      body: MyMetricsQueryHelper.constructQuery(
+        'qid',
+        MyMetricsQueryHelper.contextToTags(apiSettings.metric.context),
+        apiSettings.metric
+      )
+    },
+    function (res) {
+      $scope.metadata = res;
+      $scope.sliderBarPositionRefresh = LogViewerStore.getState().startTime;
+      console.log('Polled Response: ', res);
+      $scope.initialize();
+      if (res.status === 'KILLED') {
```

Comment:
KILLED is not the only status to stop the poll. There are COMPLETED and FAILED as well